### ISSUE TITLE
Add Multi-Label Learning to Rank (MLLTR) feature (Part 2: optimization setup)

### ISSUE THREAD

**Issue by xueyinya** [microsoft/LightGBM#5364]
## Summary
Our team developed a multi-label learning to rank (MLLTR) framework, which allows us to explore multiple different multi-objective optimization methods to build models that optimizes MLLTR with various preferred balance among different labels / objectives. We would like to open source the MLLTR framework, which is developed based on LightGBM. 

## Motivation
Learning to Rank (LTR) technique is widely used these days, especially in the Search Ranking application. It is desirable to train a model using many relevance criteria instead of just one. Therefore, our team leveraged the Multi-Objective Optimization (MOO) aspect of the MLLTR problem and use the MOO algorithms to solve MLLTR problems. With the MLLTR feature adding into the LightGBM, it would inspire new ideas in this field as well as benefit the related applications.

## Description
In Part 2, optimizers are modified and Multi-Gradient combination are introduced to further aid the MLLTR model calculation. 
Note: The user needs to specify several multi-objective (MO) related parameters in the config file to use this feature.
